src/sagas/pushNotification.js

```diff
@@ -0,0 +1,406 @@
+  yield put(pushInit({
+    deviceId,
+    settings: {
+      enabled,
+      showAmountEnabled,
+    },
+    enabledAt,
+  }));
+
+  // Check if the last registration call was made more then a week ago
+  if (enabledAt) {
+    const timeSinceLastRegistration = moment().diff(enabledAt, 'weeks');
+    if (timeSinceLastRegistration > 1) {
+      // Update the registration, as per Firebase's recommendation
+      yield put(pushRegistrationRequested({
+        enabled,
+        showAmountEnabled,
+        deviceId,
+      }));
+    }
+  }
+}
+
+/**
+ * It is responsible for loading the wallet for every push notification action that requires
+ * to interact with the wallet service api. When the user is using the facade wallet, it will
+ * load the wallet service, otherwise it will use the wallet already loaded in the redux store.
+ */
+export function* loadWallet() {
+  // This is a work-around so we can dispatch actions from inside callbacks.
+  let dispatch;
+  yield put((_dispatch) => {
+    dispatch = _dispatch;
+  });
+
+  const useWalletService = yield select((state) => state.useWalletService);
+
+  // If the user is not using the wallet-service,
+  // we need to initialize the wallet on the wallet-service first
+  let walletService;
+  if (!useWalletService) {
+    // Set urls for wallet service
+    config.setWalletServiceBaseUrl(WALLET_SERVICE_MAINNET_BASE_URL);
+    config.setWalletServiceBaseWsUrl(WALLET_SERVICE_MAINNET_BASE_WS_URL);
+    const network = new Network(NETWORK);
+
+    const pin = yield call(showPinScreenForResult, dispatch);
+    walletService = new HathorWalletServiceWallet({
+      seed: walletUtil.getWalletWords(pin),
+      network,
+      enableWs: false,
+    });
+
+    try {
+      yield call(walletService.start.bind(walletService), {
+        pinCode: pin,
+        password: pin,
+      });
+    } catch (error) {
+      yield put(pushLoadWalletFailed({ error }));
+    }
+  } else {
+    walletService = yield select((state) => state.wallet);
+  }
+
+  yield put(pushLoadWalletSuccess({ walletService }));
+}
+
+/**
+ * This function is responsible for registering the device on the wallet-service in the event
+ * of renewing the registration.
+ */
+export function* registration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+    return;
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.registerDevice, walletService, {
+      pushProvider: Platform.OS,
+      deviceId,
+      enablePush: !!enabled,
+      enableShowAmounts: !!showAmountEnabled,
+    });
+
+    if (success) {
+      const enabledAt = Date.now();
```

Comment:
If `enabled` is `false` we shouldn't update this variable in the local storage, right? Or maybe update to `0`.

Reply:
Good point!